UKHomeOffice/sgmr-service
# Spcr 490 refactor page structure

### AC / Description of changes

- create PageContainer to hold layout, ErrorSummary, LoadingSpinner
- add a 500 error handler to getData hook
- update pages to use new PageContainer & 500 error
- - Voyage departure
- - Voyage arrival
- - Voyage people selector
- - Voyage people manifest
- - Voyage employee check (is anyone employed)
- - Voyage employee select (select who is employed)

 


### To Test (Developer)

in the getData hook, change the `.then(() => {})` response to be 
```
.then(() => {
  return ({
    errors: true, status: 500, message: 'There has been a problem please try again later or view the help page', id: 'helpError',
  });
});
```

- view the page
- > the page should show the page level error with help text rather than page data
 _this is the easiest way to see visually what will happen as it is very hard to mock a 500 error to the browser_

- change it back
- run tests
- > there should be a 500 test for each page that has been updated and it should pass


### Notes

**Skipped (reason in bullet)**

- voyage pleasure craft : needs refactor to a single component to make this work properly
- voyage cya : complex page needs thinking

**Not done yet**

- Pleasure craft page
- Pleasure craft form 
- People page
- People form
- Account page
- Account form
- Dashboard
- Manage your voyage reports
- POST/PATCH/DELETE calls

---

**Remember to merge to feature branches not `master`.**
